Tools/clinic/clinic.py

```diff
@@ -530,12 +530,12 @@ class PythonLanguage(Language):
     checksum_line = "#/*[{dsl_name} end generated code: {arguments}]*/"
 
 
-ParamGroup = Iterable["Parameter"]
+ParamIter = Iterable["Parameter"]
```

Comment:
This isn't strictly needed for this PR, but after trying to use ParamGroup in `render_option_group_parsing()` (and quickly realising that was not possible), I think Param*Iter* is a more precise name for this alias.

Reply:
I agree that it's more precise. But I think it also maybe makes it harder to understand, at a glance, what the type alias _means_.

It's more verbose, but perhaps we should consider doing away with the type alias, and replacing its uses with `Iterable[Parameter]` everywhere?

Reply:
Yeah, I thought about that as well, _but_ the alias is nice for the three functions where it's used. I'm torn :)

Reply:
I would personally rip out the type alias, but what you have now is also fine. Up to you :)

Reply:
I ripped it out!